data-prepper-plugins/parse-json-processor/src/main/java/com/amazon/dataprepper/plugins/processor/parsejson/ParseJsonProcessorConfig.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.parsejson;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+
+public class ParseJsonProcessorConfig {
+    static final String DEFAULT_SOURCE = "message";
+
+    @NotEmpty
+    @NotNull
+    @JsonProperty("source")
+    private String source = DEFAULT_SOURCE;
+
+    @JsonProperty("destination")
+    private String destination;
+
+    /**
+     * The field of the Event that contains the JSON data.
+     *
+     * @return The name of the source field.
+     */
+    public String getSource() {
+        return source;
+    }
+
+    /**
+     * The destination that the parsed JSON is written to. Defaults to the root of the Event.
+     * If the destination field already exists, it will be overwritten.
+     *
+     * @return The name of the destination field.
+     */
+    public String getDestination() {
```

Comment:
I think it would be better for the default `destination` (Event root) to be `/` instead of `null`. Then we avoid some null values and my opinion is that `/` is more declarative. What do others think?

Reply:
If we used `/` as the root what would the expected behavior behavior be if someone had multiple `/`s in the destination? One might expect it to create a nested field similar to directory syntax . 

ex: Would `destination: /foo/bar` become `foo.bar: <parsed_json>` where `bar` is a nested field?

That might be useful, but I don't think that's what we're going for here so maybe `/` would be misleading. I do think your point about a more declarative default is valid though

Reply:
Right now the behavior if someone had multiple `/` in the destination (`/foo/bar`) is that the parsed json will be written to the most nested field in the destination. So if the json is `{"key":"value"}` and the destination is (`/foo/bar`), `event.get("/foo/bar", Map.class) == {"key":"value"}` and `event.get("/foo/bar/key", String.class) == "value"`. Right now the `parse_json` processor doesn't flatten nested fields (so `/foo/bar` is not flattened to `foo.bar`), it just puts any nested json into the event without changing it. 

I do see how `/` is misleading. It's also misleading for another reason: calling `event.put("/", value)` will not put `value` into the event root because the `JacksonEvent` implementation trims leading slashes. So it would not write `value` to the event at all. 

I think we can stick with a `null` default and be more declarative by mentioning that `destination is null -> write to root` in documentation.